### PR TITLE
Update actions versions to Node 20

### PR DESCRIPTION
**Pull request by sciencewhiz**

### CODE REVIEW

File: .github/workflows/dist.yml [hunk 1]
```diff
@@ -123,7 +123,7 @@ jobs:
         RPYBUILD_STRIP_LIBPYTHON: "1"
         RPYBUILD_CC_LAUNCHER: ccache
 
-    - uses: actions/upload-artifact@v3
+    - uses: actions/upload-artifact@v4
       with:
         name: "dist-${{ runner.os }}"
```
Comment (auscompgeek) [inline .github/workflows/dist.yml:128]: This won't work. This workflow is currently relying on the merging behaviour in artifacts v3.

**Review by auscompgeek — DISMISSED** [review]

File: .github/workflows/dist.yml (hunk 1 above)
Comment (sciencewhiz) [reply .github/workflows/dist.yml:128]: Whoops, I saw there was names for each but didn't realize they weren't unique.

Comment (virtuald) [reply .github/workflows/dist.yml]: Switching the workflow will also require changing the way docs are built, currently RTD downloads the docs from github actions via https://github.com/robotpy/robotpy-sphinx-plugin/blob/main/robotpy_sphinx/download_and_install.py

Comment (sciencewhiz) [reply .github/workflows/dist.yml]: Assuming I did the merging correctly, the artifacts should have the same name and contents, so it looks like that should still work, unless I'm missing something. There are temporary artifacts created, but they are deleted after merging.

Comment (sciencewhiz) [reply .github/workflows/dist.yml]: It looks to me like the artifacts are the same as before.

**Pull request merged**